drv/local-vpd/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
A top-level block comment about what this is for or why it exists would be nice. But I can appreciate that may not be the hubris style. Just coming to this as an outsider it's not really clear what this is or isn't for.

Are we going to have a copy of this driver for each different i2c source in a board based on the build.rs?

What makes something "local"? I guess that it's on the same physical board as the SP and not a FRU like the fan modules or sharkfins?

Reply:
Added a top-level comment explaining that this is the FRU ID EEPROM soldered to the board.

> Are we going to have a copy of this driver for each different i2c source in a board based on the build.rs?

I'm not sure what this means – each task that wants to read from the FRU ID will include this driver (as a library) and use it to extract the tags that the task cares about.

Reply:
> Added a top-level comment explaining that this is the FRU ID EEPROM soldered to the board.

Thank you. I guess it's a little weird that we'd rebuild this and use the same driver as a library in everything and are calling it local when the other's aren't local, but that's probably just me.

> > Are we going to have a copy of this driver for each different i2c source in a board based on the build.rs?
> 
> I'm not sure what this means – each task that wants to read from the FRU ID will include this driver (as a library) and use it to extract the tags that the task cares about.

So I was trying to sort through how this worked in practice. When the SP ultimately wants to do the same thing with the 12 different FRUID ROMs that we have, it sounds like we'll build this all (not just the build.rs) 12 different times even though the only difference is generally just which i2c to scan for the FRU block for and get the corresponding tagged data out of. Not knowing more about the overall design of the system that feels surprising, but if that's the general pattern then it is.